app/models/fe/reference_sheet.rb

```diff
@@ -76,7 +76,8 @@ class ReferenceSheet < ActiveRecord::Base
 
     alias_method :applicant, :applicant_answer_sheet
     def generate_access_key
-      self.access_key = Digest::MD5.hexdigest(email + Time.now.to_s)
+      @@nonce ||= 0
+      self.access_key = Digest::MD5.hexdigest(email + Time.now.strftime("%G-W%V-%uT%T%:z%L") + Fe::ReferenceSheet.maximum(:id).to_s + (@@nonce += 1).to_s)
```

Comment:
Yeah I like that better too. Simpler. I'll update this later with it.
On 24 Jun 2015 21:20, "Josh Starcher" <notifications@github.com> wrote:

> In app/models/fe/reference_sheet.rb
> <https://github.com/CruGlobal/qe/pull/28#discussion_r33214814>:
>
> > @@ -76,7 +76,8 @@ class ReferenceSheet < ActiveRecord::Base
> >
> >      alias_method :applicant, :applicant_answer_sheet
> >      def generate_access_key
> > -      self.access_key = Digest::MD5.hexdigest(email + Time.now.to_s)
> > +      @@nonce ||= 0
> > +      self.access_key = Digest::MD5.hexdigest(email + Time.now.strftime("%G-W%V-%uT%T%:z%L") + Fe::ReferenceSheet.maximum(:id).to_s + (@@nonce += 1).to_s)
>
> I prefer this pattern:
> https://github.com/CruGlobal/sp/blob/394c1ddf458a7ff25bdca361cb7810017f96be84/app/models/api_key.rb#L6
>
> What you did is definitely a big improvement, but that pattern will be 100%
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CruGlobal/qe/pull/28/files#r33214814>.
>